aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -3789,7 +3790,49 @@ public SignInProviderConfig(final Class<? extends SignInProvider> signInProvider
     }
 }
 
-class AWSMobileClientStore {
+interface KeyValueStore {
+    Map<String, String> get(final String... keys);
+    String get(final String key);
+    void set(final Map<String, String> attributes);
+    void set(final String key, final String value);
+    void clear();
+}
+
+final class DummyStore implements KeyValueStore {
+    @Override
+    public Map<String, String> get(String... keys) {
+        throwNotInitializedException();
+        return null;
+    }
+
+    @Override
+    public String get(String key) {
+        throwNotInitializedException();
+        return null;
+    }
+
+    @Override
+    public void set(Map<String, String> attributes) {
+        throwNotInitializedException();
+    }
+
+    @Override
+    public void set(String key, String value) {
+        throwNotInitializedException();
+
+    }
+
+    @Override
+    public void clear() {
+        throwNotInitializedException();
+    }
+
+    private void throwNotInitializedException() {
+        throw new RuntimeException("AWSMobileClient has not been initialized yet.");
```

Comment:
So, it looks like this PR doesn't fix the crash, it just replaces the NPE with this more specific crash, which could help us to identify the real root cause later, is that right?

Reply:
yep. the customer will still come across an error, but we'll know if it was caused by faulty initialization or something else.